test/join.jl

```diff
@@ -1557,4 +1559,163 @@ end
                     c="c", d="d")
 end
 
+@testset "matchmissing :notequal correctness" begin
+    Random.seed!(1337)
+    names = [
+        DataFrame(ID = Union{Int, Missing}[1, 2, missing],
+            Name = Union{String, Missing}["John Doe", "Jane Doe", "Joe Blogs"]),
```

Comment:
Align with `ID`.

Reply:
while we are at it - we do not put spaces for kwargs, so e.g. it should be `DataFrame(ID=Union{Int, Missing}[], Name=String[])`

Reply:
I tried to mimic other tests, what is preferred by the way?
1)
```julia
DataFrame(ID=[1:100; missings(100)],
          Name=repeat(["Jane Doe"], 200)),
```
or
2)
```julia
DataFrame(ID=Union{Int, Missing}[1:100; missings(100)],
          Name=Union{String, Missing}[repeat(["Jane Doe"], 200)...]),
```
?

Reply:
```
DataFrame(ID=[1:100; missings(100)],
          Name=repeat(["Jane Doe"], 200))
```
is preferred unless you want to explicitly test what happens if you have a column that does not have `missing`, but has `Missing` in `eltype` union. As I have commented in this case actually it has some impact as `completecases` is sensitive to this difference (and it is called at some point of the processing), but since we test `completecases` elsewhere we do not need to do it here.

Reply:
Ok, makes a perfect sense. I felt like we want to write explicit types when possible for some reason (that is important, but I don't get it yet :D).